spec/shared_examples/checkout.rb

```diff
@@ -0,0 +1,239 @@
+shared_examples 'displays info' do |name|
+  it name do
+    # customer_email = find('#order_email').value
+
+    aggregate_failures do
+      expect(page).to have_css('div[data-hook="checkout_header"]')
+      expect(page).to have_css('.first', text: 'Address')
+      expect(page).to have_css('.next', text: 'Delivery')
+      expect(page).to have_css('li', text: 'Payment')
+      expect(page).to have_css('.last', text: 'Complete')
+
+      expect(page).to have_css('.form-group', text: 'Customer E-Mail')
+      # expect(customer_email).to eq ENV['USERNAME_SPREE']
+
+      expect(page).to have_css('.panel-heading', text: 'Billing Address')
+      expect(page).to have_css('.panel-heading', text: 'Shipping Address')
+      expect(page).to have_css('#checkout-summary', text: 'Order Summary')
+
+      expect(page).to have_button('Save and Continue')
+    end
+  end
+end
+
+shared_examples 'default country' do |name|
+  it name do
+    uncheck 'order_use_billing'
+
+    aggregate_failures do
+      expect(page).to have_css('#order_bill_address_attributes_country_id', text: 'United States of America')
+      expect(page).to have_css('#order_ship_address_attributes_country_id', text: 'United States of America')
+    end
+  end
+end
+
+shared_examples 'save address' do |name|
+  it name do
+    fill_in_billing(address)
+
+    if name.include?('Shipping')
+      uncheck 'order_use_billing'
+      fill_in_shipping(address2)
+    end
+
+    click_button 'Save and Continue'
+
+    expect(page).to have_css('.active', text: 'Delivery')
+  end
+end
+
+shared_examples 'has all needed info' do |name|
+  it name do
+    shipping_method = all('.rate-cost')
+    ups_ground = shipping_method[0].text
+    ups_two_day = shipping_method[1].text
+    ups_one_day = shipping_method[2].text
+
+    aggregate_failures do
+      expect(page).to have_css('.rate-name', text: 'UPS Ground (USD)')
+      expect(ups_ground).to eq '$5.00'
+      expect(page).to have_css('.rate-name', text: 'UPS Two Day (USD)')
+      expect(ups_two_day).to eq '$10.00'
+      expect(page).to have_css('.rate-name', text: 'UPS One Day (USD)')
+      expect(ups_one_day).to eq '$15.00'
+      expect(page).to have_css('.item-name', text: 'Ruby on Rails Bag')
+      expect(page).to have_css('.item-name', text: 'Ruby on Rails Tote')
+    end
+  end
+end
+
+shared_examples 'shipping methods' do |name|
+  it name do
+    choose_shipping_method('UPS One Day (USD)', '$15.00')
+    expect(page).to have_css('tr[data-hook="shipping_total"]', text: '$15.00')
+
+    choose_shipping_method('UPS Ground (USD)', '$5.00')
+    expect(page).to have_css('tr[data-hook="shipping_total"]', text: '$5.00')
+
+    choose_shipping_method('UPS Two Day (USD)', '$10.00')
+    expect(page).to have_css('tr[data-hook="shipping_total"]', text: '$10.00')
+
+    if name.include?('save')
```

Comment:
maybe you should make it as 1 test instead of making the same steps on both test cases?